### ISSUE TITLE
General suggestion thread

### ISSUE THREAD

**Issue by ghost** [Saishy/certainmangareader#8]
Partially from old thread.

Issues:
- Compact mode occosionally slowing down browsers to a crawl.
- Chapters deleted from manga sites after you read them cause the update system to hang until manga is re-added.
- Unstable on 32-bit Firefox. Might be caused by the memory leak.

Suggestions and requests:
- Mangacow, Kissmanga and Crunchyroll support.
- Fixing Batoto's symbol bug.
- A button next to the blue arrow button to go to the next chapter
- Categories
- A permanent counter for how many series with new chapters are in the list.
- Sorting options.
- A way to sync with a MyAnimeList account

**Comment by Saishy** [comment]
About the manga's name, is a bug at Batoto, I will fix it internally in my addon tho.
- A permanent counter will be implemented as an option (priority: very low)
- A button to go to the next chapter (priority: low)
- Sync with a MyAnimeList (priority: none, will only be implemented in a unlikable future)
- Sorting options (priority: very low)
- A way to differentiate between "unfinished" and "newly updated manga" (priority: none, this is a personal question of what defines a newly updated manga and just an unfinished one, the addon has no way to know when the chapter was released, only when it received an update. A manga that you haven't finished is always considered an unfinished manga and is a logical parameter. This also adds another layer of information to the user that is distracting)
- Mangacow (priority: very high)
- Kissmanga (priority: very high)
- Crunchyroll (priority: medium)

**Comment by ghost** [comment]
Much appreciated. I see the issue with the "unfinished vs updated" thing, but how about categories? Or, for example, being able to mark unfinished manga yourself that will be shown below unmarked ones. As it is I do not have my unfinished manga in the plugin, as the new chapter get lost among them.

**Comment by Keats** [comment]
import/export would be necessary, ie if i want to reinstall my laptop for example.
Can be just json in/out

**Comment by Saishy** [comment]
@Keats There is already an import/export at the latest beta version.

**Comment by Keats** [comment]
Can't believe I missed it, thanks!

**Comment by Saishy** [comment]
@PeFoHe I'm putting the categories thing into "sorting options" and moving the priority to low.

**Comment by ghost** [comment]
Thanks.

**Comment by PeFoHe** [comment]
I cleaned up and updated this.

**Comment by ghost** [comment]
I tested a bit more. The two update issues(Merged above) did not appear until I reset my pc and imported the manga list. I now went into every manga on my list and pressed the "Mark current chapter as latest read" button on all of them. This not only allowed them to seemengly permanently update again, but also allowed the "update now" button to turn back to normal after a update.

I'm assuming there is a bug in the import/export function that can make the updates get stuck on certain manga until manually updated.

**Comment by Saishy** [comment]
Uhm, I must say I have never ever encountered that problem and never heard of it...
I will try to look into it but I have no clue what the hell is going wrong at those last versions :c

**Comment by ghost** [comment]
Thanks, not a big problem though. Once the troublesome manga have been re-added there is no problem. I can reproduce it by importing [my manga list](https://github.com/Saishy/certainmangareader/files/183006/Export.txt) and trying to update.

**Comment by Saishy** [comment]
Just to update: I'm working at two jobs right now, and both at nearly release so It will take a little time before I get back to working at my addon.

**Comment by ghost** [comment]
No problem, take your time. Good luck on whatever you're working on.

**Comment by ghost** [comment]
~~Turns out that I was using the 32-bit version of Firefox, the compact mode works better on 64-bit. I'm not sure if it's a problem on your side or Mozillas side, but this plugin is very unstable on the 32-bit browser, making it hit the 2GB ram limit for no apparent reasons. I can elaborate if you want, but then again, who intentionally uses 32-bit browsers?~~

EDIT: Incorrect, other than instability on 32-bit.

**Comment by ghost** [comment]
Alright, compact mode stopped working again. Still using the 64-bit browser. I no longer have any idea what's going on.
When opening the browser while the plugin is in compact mode firefox uses 30% of my cpu and does not react to anything, normal cpu usage seems to be at 1-10%.
This is on a Surface Pro 4 running Windows 10 (my Macbook has been scrapped). The memory leak you have a thread about is also noticable here, unlike on my desktop or Macbook.

TL:DR: Compact mode works sometimes.

**Comment by ghost** [comment]
As it turns out, the updating does _not_ get stuck because of imports - my manga list just contained an error. The actual cause seems to be links that no longer exist. 

The update now got stuck for me because of [Nanatsu no Taizai chapter 135.5](http://bato.to/reader#4d4d8caf5736bf9b), which was the latest chapter I had read before it was deleted. That link is broken. Removing the manga fixed the updating.

Updating thread.

**Comment by Saishy** [comment]
Closing this issue as being outdated, please create another if problem persists after new release.

**Comment by ghost** [comment]
Aw yiss, updates!
